### PR TITLE
Create auth context

### LINKED ISSUE #2: Create authContext using useContext React Hook
# Description
- In order to make a 'global' react state, we need to provide all project through a react context

#  Prerequisites
- create and configure AuthContext
- provide it wrapping all the app react component
- make a useAuth that gets a context to export

# Additional Info
- make it like our previous projects such as 'react-bombado'

### PR DESCRIPTION
**Pull request by Lucas-Marques-Leao**
# Related Issue
- #2  

# Proposed Changes
- made the issue
- added sweetalert2 and axios

# Checklist
- [ ] Tests
- [ ] Translations
- [ ] Documentation

### CODE REVIEW

**Comment by Lucas-Marques-Leao** [comment]
PS: Label on hold doesn't exist, but it's an on hold pull request

File: src/contexts/AuthContext.tsx [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+  register: (
+    name: string,
+    email: string,
+    password: string,
+    passwordConfirmation: string
+  ) => Promise<void>;
+  login: (email: string, password: string) => Promise<void>;
+  logout: () => Promise<void>;
+}
+
```
Comment (duardoqueiroz) [inline src/contexts/AuthContext.tsx]: I don't know about good practices in react, but wouldn't it be interesting to create a directory just for the interfaces?

File: src/contexts/AuthContext.tsx [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+export const AuthContext = createContext<AuthContextData>(
+  {} as AuthContextData
+);
+
+interface AuthProviderProps {
+  children: React.ReactNode;
+}
```
Comment (duardoqueiroz) [inline src/contexts/AuthContext.tsx]: here too?

File: src/services/api/index.ts [hunk 3]
```diff
@@ -0,0 +1,7 @@
+import axios from "axios";
+
+const api = axios.create({
+  baseURL: "http://localhost:3333",
```
Comment (duardoqueiroz) [inline src/services/api/index.ts]: it is interesting that this value is in an environment variable

**Review by duardoqueiroz — APPROVED** [review]
please check my suggestions and if they make sense implement them.

File: src/services/api/index.ts (hunk 3 above)
Comment (Lucas-Marques-Leao) [reply src/services/api/index.ts]: Doing that posthaste

File: src/contexts/AuthContext.tsx (hunk 1 above)
Comment (Lucas-Marques-Leao) [reply src/contexts/AuthContext.tsx]: I'll make a research

File: src/contexts/AuthContext.tsx [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+import {
+  IAuthContextData,
+  IAuthProviderProps,
+} from "../shared/interfaces/IAuthContext";
+
+export const AuthContext = createContext<IAuthContextData>(
+  {} as IAuthContextData
+);
+
+const MySwal = withReactContent(Swal);
```
Comment (Manuel-Antunes) [inline src/contexts/AuthContext.tsx]: This swal can become a service too, create a file called `swal.ts` in services folder, and export this HOC swal as singleton

File: src/services/api/index.ts [hunk 5]
```diff
@@ -0,0 +1,7 @@
+import axios from "axios";
+
+const api = axios.create({
+  baseURL: process.env.SERVER_URL,
```
Comment (Manuel-Antunes) [inline src/services/api/index.ts]: This env really works? we're using vite, and work with environment variables is a quite different
check it out: https://vitejs.dev/guide/env-and-mode.html

**Review by Manuel-Antunes — CHANGES_REQUESTED** [review]
Check my issues, solve all problems, then, request my review again

**Review by Manuel-Antunes — APPROVED** [review]
OK!

**Pull request merged**